tests/test_algorithms.py

```diff
@@ -17,8 +17,7 @@ def bob_ross(test_examples, filename=BOB_ROSS):
 
     context = concepts.load_cxt(str(path), encoding=ENCODING)
 
-    assert len(context.objects) == 403
-    assert len(context.properties) == 67
+    assert context.shape == (403, 67)
```

Comment:
In case you have grep or something similar handy, you should find a couple more of these in `scripts/` and `tests/` with a regex such as `len\(.*\.(object|propertie)s\)`

Reply:
Originally I kept these in `scripts/` without changes for better readability (and :_d formatting), mainly in case of normal `Tuple` and not `NamedTuple`. I have not seen any more of them in `tests/`.

In case of `NamedTuple` version, i see, that replacing them everywhere is more logical (no accessing with index):

Do you want me to keep `:_d` formatting?
`assert context.shape == (403, 67), f'({context.shape.rows:_d}, {context.shape.columns:_d}) != (403, 67)'`

The simpler version would be:
`assert context.shape == (403, 67), f'{context.shape} != (403, 67)'`

Reply:
Thanks. SG, let's save a line (also in the `scripts/` above).

Reply:
Fine with me to drop the assertion messages with `:_d` formatting